devito/types/dimension.py

```diff
@@ -890,7 +900,7 @@ def _arg_defaults(self, _min=None, size=None, alias=None):
             return defaults
         try:
             # Is it a symbolic factor?
-            factor = defaults[dim._factor.name] = dim._factor.data
+            factor = defaults[dim._factor.name] = self._factor.data
```

Comment:
why?

Reply:
same as above, self is the runtime argument with the value for the operator, dim is the name of the dimension used to build the operator and argument name